mockserver/src/main/groovy/pl/touk/mockserver/server/ContextExecutor.groovy

```diff
@@ -92,4 +93,22 @@ class ContextExecutor {
     List<Mock> getMocks() {
         return mocks
     }
+
+    private void handleMaxUses(Mock mock) {
+        if (mock.hasLimitedUses()) {
+            mock.decrementUses()
+            removeAndResetIfNeeded(mock)
+            log.debug("Uses left ${mock.usesLeft} of ${mock.maxUses} (is cyclic: ${mock.cyclic})")
+        }
+    }
+
+    private synchronized void removeAndResetIfNeeded(Mock mock) {
```

Comment:
synchronizing this metod is not enough...
if we want to synchronize we should synchronize `handleMaxUses`.
Additionally we should consider do we really need synchronize on `this`, maybe we could synchronize on `mock`?

Reply:
Fixed.